README.md

```diff
@@ -46,12 +46,128 @@ The `main` and `db` packages contain integration tests that require a successful
 ### No-scan domains
 In case of complaints or abuse, we may not want to continually scan some domains. You can set the environment variable `DOMAIN_BLACKLIST` to point to a file with a list of newline-separated domains. Attempting to scan those domains from the public-facing website will result in error codes.
 
-## API
+## Scan API
 
+Our API objects can look a bit complicated! There's lots of information contained in a TLS scan.
 To request a scan:
 ```
 POST /api/scan
+  { "domain": "example.com" }
+```
+
+SLet's break down exactly what each part of this giant nested response means. All API responses, not just scans, are wrapped in a JSON, like:
+```
+{
+    status_code: 200,
+    message: "",
+    response: <reponse data>
+}
+```
+Or even:
+```
+{
+    status_code: 400,
+    message: "query parameter domain not specified",
+    response: {}
+}
+```
+The status codes correspond with regular HTTP errors, and `message` provides more context into particular failures.
+
+### Scan responses
+
+Here's an abbreviated scan response. There's extra information on these objects that help
+describe the errors we encountered.
+```
+{
+    domain: "example.com",
+    scandata: {
+        status: 0,
+        results: {  // Individual hostname check results
+            "mx.example.com": {
+                "status": 0,
+                "checks": {
+                    "connectivity": { "status": 0 },
+                    "certificate": { "status": 0 },
+                    "starttls": { "status": 0 },
+                    "version": { "status": 0 },
+                }
+            }
+            "dummy.example.com": {
+                "status": 3,
+                "checks": {
+                    "connectivity": {
+                        "status": 3,
+                        "messages": [ "Error: Could not establish connection" ]
+                    },
+                }
+            },
+        },
+        preferred_hostnames: ["mx.example.com"], // Hostnames we were able to connect to
+        extra_results: {"policylist": { "status": 0 }},
+    },
+    timestamp: 0,
+    version: 1,
+}
+```
+
+The meat of the response is in scandata, which is a JSON-ification of the `DomainResult` structure returned from the `checker` package.
+
+### Domain results
+
+Here's a quick synopsis of the top-level fields:
```

Comment:
How are these fields "top-level", since some of them occur in more deeply nested objects than others?

Should `timestamp` and `version` also be documented here?